src/Cassandra/ControlConnection.cs

```diff
@@ -538,29 +538,28 @@ public IEnumerable<Row> Query(string cqlQuery, bool retry = false)
             return TaskHelper.WaitToComplete(QueryAsync(cqlQuery, retry), MetadataAbortTimeout);
         }
 
-        public Task<IEnumerable<Row>> QueryAsync(string cqlQuery, bool retry = false)
+        public async Task<IEnumerable<Row>> QueryAsync(string cqlQuery, bool retry = false)
         {
             var request = new QueryRequest(ProtocolVersion, cqlQuery, false, QueryProtocolOptions.Default);
-            var task = _connection
-                .Send(request)
-                .ContinueSync(GetRowSet);
-            if (!retry)
+            Response response;
+            try
             {
-                return task;
+                response = await _connection.Send(request).ConfigureAwait(false);
             }
-            return task.ContinueWith(t =>
+            catch (SocketException ex)
             {
-                var ex = t.Exception != null ? t.Exception.InnerException : null;
-                if (ex is SocketException)
+                const string message = "There was an error while executing on the host {0} the query '{1}'";
+                _logger.Error(string.Format(message, cqlQuery, _connection.Address), ex);
```

Comment:
message is not reused by another string.Format, so it could change to use string interpolation